sklearn/tree/_criterion.pyx

```diff
@@ -681,25 +754,78 @@ cdef class RegressionCriterion(Criterion):
         self.reset()
         return 0
 
+    cdef void init_missing(self, SIZE_t n_missing) nogil:
+        """Initalize sum_missing if there are missing values.
+
+        This method assumes that caller placed the missing samples in
+        self.samples[-n_missing:]
+        """
+        cdef SIZE_t i, p, k
+        cdef DOUBLE_t* sample_weight = self.sample_weight
+        cdef SIZE_t* samples = self.samples
+        cdef DOUBLE_t w
+        cdef DOUBLE_t y_ik
+        cdef DOUBLE_t w_y_ik
+
+        self.n_missing = n_missing
+        if n_missing == 0:
+            return
+
+        memset(&self.sum_missing[0], 0, self.n_outputs * sizeof(double))
+
+        self.weighted_n_missing = 0.0
+        for p in range(self.end - n_missing, self.end):
```

Comment:
In the docstring, for `init_missing` it does note that the missing values should be at the end. This happened because the splitter placed the missing values at the end before sorting.

I'll add another comment here.